Frogies-Project/Assets/Scripts/Core/PotionSystem/PotionSystem.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using Items.Core;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Core.PotionSystem
+{
+    public class PotionSystem : MonoBehaviour
+    {
+        [SerializeField] private GameObject optionPrefab;
+        [SerializeField] private Transform body;
+        private List<PotionOption> _options;
+        public event EventHandler OnOptionSelected;
+        public event EventHandler OnActive;
+        
+        public void Setup(List<Potion> potions)
+        {
+            _options = new List<PotionOption>();
+            foreach (var potion in potions)
+            {
+                GameObject potionGameObject = Instantiate(optionPrefab, body, true);
+                PotionOption option = potionGameObject.GetComponent<PotionOption>();
+                option.Setup(potion);
+                option.OnSelected += OptionOnSelected;
+                _options.Add(option);
+            }
+            gameObject.SetActive(false);
+        }
+
+        public void OnNewDay() //connect with day/night feature
```

Comment:
Hmm not sure if its a good naming. We usually use 'On' in events. I would rather rename it to OpenPotionMenu.